### PR TITLE
Added `reset` command.

### PR DESCRIPTION
**Pull request by bpeabody**

### CODE REVIEW

File: node/lib/cmd/reset.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    let commit;
+    if (args.commitish) {
+
+        const commitish = args.commitish;
+        const annotated = yield GitUtil.resolveCommitish(repo, commitish);
+
+        if (null === annotated) {
+            throw new UserError(`Could not resolve ${colors.red(commitish)}.`);
+        }
+        commit = yield repo.getCommit(annotated.id());
```
Comment (bpeabody) [inline node/lib/cmd/reset.js]: This method does throw (I wish it returned null/error instead), but I think we can assume that the fact that the `AnnotatedCommit` is valid (i.e., we resolve the comitish) implies the existence of a valid commit.

File: node/lib/cmd/reset.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    }
+    else {
+        commit = yield repo.getHeadCommit();
+    }
+
+    // Perform the reset.
+
+    let type = Reset.TYPE.MIXED;
+
+    if (args.soft) {
```
Comment (bpeabody) [inline node/lib/cmd/reset.js]: Yes, I agree.

**Comment by bpeabody** [comment]
Think I got everything.

File: node/lib/cmd/reset.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        action: "storeConst",
+        constant: true,
+        help: `Do not change the working directory or index.`,
+    });
+
+    parser.addArgument(["--mixed"], {
+        required: false,
+        action: "storeConst",
+        constant: true,
+        help: `Unstage any changes, but to not alter the working dictory.  \
```
Comment (novalis) [inline node/lib/cmd/reset.js]: s/to not/do not/

Comment (bpeabody) [inline node/lib/cmd/reset.js]: got it, thanks

**Pull request merged**